core/src/xmake/readline/clear_history.c

```diff
@@ -0,0 +1,59 @@
+/*!The Make-like Build Utility based on Lua
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright (C) 2015 - 2017, TBOOX Open Source Group.
+ *
+ * @author      TitanSnow
+ * @file        clear_history.c
+ *
+ */
+
+/* //////////////////////////////////////////////////////////////////////////////////////
+ * trace
+ */
+#define TB_TRACE_MODULE_NAME                "clear_history"
+#define TB_TRACE_MODULE_DEBUG               (0)
+
+/* //////////////////////////////////////////////////////////////////////////////////////
+ * includes
+ */
+#include "prefix.h"
+
+#ifdef XM_CONFIG_API_HAVE_READLINE
+
+/* //////////////////////////////////////////////////////////////////////////////////////
+ * implementation
+ */
+
+// clear_history wrapper
+tb_int_t xm_readline_clear_history(lua_State* lua)
+{
+    // check
+    tb_assert_and_check_return_val(lua, 0);
+
+    // call clear_history
+    clear_history();
```

Comment:
It will crash on macOS here.

```bash
ruki:~ ruki$ lldb /usr/local/share/xmake/xmake 
(lldb) target create "/usr/local/share/xmake/xmake"
Current executable set to '/usr/local/share/xmake/xmake' (x86_64).
(lldb) run lua
Process 32656 launched: '/usr/local/share/xmake/xmake' (x86_64)
Process 32656 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x00007fffa48c6e13 libedit.3.dylib`history + 912
libedit.3.dylib`history:
->  0x7fffa48c6e13 <+912>: movq   (%r13), %rdi
    0x7fffa48c6e17 <+916>: movq   %r15, %rsi
    0x7fffa48c6e1a <+919>: callq  *0x48(%r13)
    0x7fffa48c6e1e <+923>: xorl   %r15d, %r15d
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
  * frame #0: 0x00007fffa48c6e13 libedit.3.dylib`history + 912
    frame #1: 0x00007fffa48bc569 libedit.3.dylib`clear_history + 31
    frame #2: 0x00000001000049dd xmake`xm_readline_clear_history(lua=0x0000000000084378) at clear_history.c:50
    frame #3: 0x00000001000a0eb6 xmake`lj_BC_FUNCC + 52
    frame #4: 0x0000000100024102 xmake`lua_pcall(L=0x0000000000084378, nargs=0, nresults=1, errfunc=-2) at lj_api.c:1052
    frame #5: 0x00000001000011b7 xmake`xm_machine_main(machine=0x0000000000400bc0, argc=2, argv=0x00007fff5fbff760) at machine.c:538
    frame #6: 0x0000000100000ad3 xmake`main(argc=2, argv=0x00007fff5fbff760) at xmake.c:22
    frame #7: 0x0000000100000a64 xmake`start + 52
(lldb) f 2
frame #2: 0x00000001000049dd xmake`xm_readline_clear_history(lua=0x0000000000084378) at clear_history.c:50
   47  	    tb_assert_and_check_return_val(lua, 0);
   48  	
   49  	    // call clear_history
-> 50  	    clear_history();
   51  	
   52  	    // pushnil
   53  	    lua_pushnil(lua);
(lldb) 

```